web/geo/validate-v2.js

```diff
@@ -0,0 +1,187 @@
+const {
+  area,
+  bbox,
+  bboxPolygon,
+  convex,
+  dissolve,
+  featureCollection,
+  featureEach,
+  getGeom,
+  intersect,
+} = require('@turf/turf');
+const { getPolygons, getHoles, writeJSON, log } = require('./utilities');
+
+const { getZonesJson } = require('./files');
+
+// TODO: Improve this function so each check returns error messages,
+// so we can show all errors instead of taking them one at a time.
+function validateGeometryV2(fc, config) {
+  const fcHighFidelity = {
+    ...fc,
+    features: fc.features.filter((feature) => {
+      try {
+        return !feature.properties.aggregatedView;
+      } catch (e) {
+        console.log('Error: ', e, 'Feature: ', feature); // eslint-disable-line no-console
+      }
+    }),
+  };
+
+  const fcAggregated = {
+    ...fc,
+    features: fc.features.filter((feature) => {
+      try {
+        return feature.properties.aggregatedView;
+      } catch (e) {
+        console.log('Error: ', e, 'Feature: ', feature); // eslint-disable-line no-console
+      }
+    }),
+  };
+  console.log('Validating geometries high fidelity view...'); // eslint-disable-line no-console
+  zeroNullGeometries(fcHighFidelity);
+  containsRequiredProperties(fcHighFidelity);
+  zeroComplexPolygons(fcHighFidelity, config);
+  zeroNeighboringIds(fcHighFidelity);
+  zeroGaps(fcHighFidelity, config);
+  zeroOverlaps(fcHighFidelity, config);
+  matchesZonesConfig(fcHighFidelity, config);
+
+  if (fcAggregated.features.length) {
+    console.log('Validating geometries aggregated view..'); // eslint-disable-line no-console
+    zeroNullGeometries(fcAggregated);
+    containsRequiredProperties(fcAggregated);
+    zeroComplexPolygons(fcAggregated, config);
+    zeroNeighboringIds(fcAggregated);
+    zeroGaps(fcAggregated, config);
+    zeroOverlaps(fcAggregated, config);
+    matchesZonesConfig(fcAggregated, config);
+  }
```

Comment:
Here we could create an inserter-function/wrapper so we don't have to duplicate all lines.

Reply:
✂️